### PR TITLE
[Go] Concepts of hamming

### LINKED ISSUE #235: [Go] Concepts of hamming (v2)
Extract a list of Go concepts needed to solve the v2 hamming exercise. Each concept should have a short info why it's needed.

Create a PR with the file languages/go/reference/exercise-concepts/hamming.md.

### PR DESCRIPTION
**Pull request by andres-zartab**
Add languages/go/reference/exercise-concepts/hamming.md
First take at capturing hamming concepts for Go.
Attempts to fix #235

### CODE REVIEW

File: languages/go/reference/exercise-concepts/hamming.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+## Concepts encountered while solving it
+
+## Strings
+Strings in Go are made up of runes. Runes are Go's equivalent to the concept of characters. A rune can be represented using from 1 up to 4 bytes. 
+
+### String iteration
+- When iterating over strings using the range operator, runes are yielded. When dealing with ASCII runes, strings can be safely ranged using the string[i] syntax. However when non ASCII characters are present this is unsafe as only one byte of the corresponing rune is returned.
+
+### String comparison
+- Since two strings cannot be ranged using the range operator at the same time, the safe approach for comparing each "character" in two strings is by first casting the strings to a slice of runes.
```
Comment (tehsphinx) [inline languages/go/reference/exercise-concepts/hamming.md]: I'd call the concept `String conversion`

File: languages/go/reference/exercise-concepts/hamming.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Errors
+In Go there are no exceptions, instead errors can be defined and processed accordingly to the context. Errors can be defined using fmt.Errorf or errors.New. The latter is preferred when there is no string interpolation within the error message. Error messages should be concise, not vague and provide context for where and why the error occurred. Also they should not be capitalized or have punctuation.
+
+When an error is encountered the convention is to return a zero value along with the corresponging error. In the opposite case a nil error should be returned along with the value that the function should return.
```
Comment (tehsphinx) [inline languages/go/reference/exercise-concepts/hamming.md]: Maybe split this up in 2 concepts: `error creation` and `returning errors`. Those 2 are something we should probably teach very early on in a concept exercise maybe called `error-basics`.

Comment (tehsphinx) [reply languages/go/reference/exercise-concepts/hamming.md]: To `returning errors` also belongs that they should always be the last return parameter. Doesn't need to mentioned here. Just mentioning it as there are a few things to teach here.

Comment (andres-zartab) [reply languages/go/reference/exercise-concepts/hamming.md]: I like the idea of having an early exercise with basic error concepts, given how common they are.

**Pull request merged**